src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -181,7 +177,7 @@ protected virtual HashAlgorithmName GetHashAlgorithmName(string algorithm)
             throw LogHelper.LogException<ArgumentOutOfRangeException>(LogMessages.IDX10640, algorithm);
         }
 
-        private void ResolveDotNetCoreAsymmetricAlgorithm(AsymmetricSecurityKey key, string algorithm, bool willCreateSignatures)
+        public virtual void ResolveAsymmetricAlgorithm(AsymmetricSecurityKey key, string algorithm, bool willCreateSignatures)
```

Comment:
Lets discuss this on https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/335